src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs

```diff
@@ -28,7 +28,7 @@ public abstract partial class SafeHandle : CriticalFinalizerObject, IDisposable
         //   code, so this managed code must not assume it is the only code
         //   manipulating _state.
 
-#if DEBUG
+#if DEBUG && CORECLR
         private readonly string? _ctorStackTrace;
 #endif
```

Comment:
@stephentoub questions:
1. Can we move this member to the end of the set of instance members. (And I assume do the related changes on the native side in coreclr)
2. For Mono we should really have `[StructLayout(Sequential)]` on any class where we care about the field layout.  Otherwise we're just relying on auto layout coincidentally getting the fields in the right order.
3. Should we make `SafeHandle.Mono.cs` and `SafeHandle.CoreCLR.cs` that just contain the instance field definitions?  `#if DEBUG && CORECLR` accomplishes the same thing but feels more fragile.

Reply:
> Can we move this member to the end of the set of instance members. (And I assume do the related changes on the native side in coreclr)

I'd previously done that, and I believe the issues I hit were to do with alignment, though it's also possible I just messed something up or neglected to edit something I needed to edit.

> For Mono we should really have [StructLayout(Sequential)] on any class where we care about the field layout. Otherwise we're just relying on auto layout coincidentally getting the fields in the right order.

This is irrespective of this change or this field, right?  You're saying mono is expecting that handle is first and it may not be.

> Should we make SafeHandle.Mono.cs and SafeHandle.CoreCLR.cs that just contain the instance field definitions? #if DEBUG && CORECLR accomplishes the same thing but feels more fragile.

I don't love the duplication there; that feels more fragile to me, actually.  If this field is problematic, we can just remove it and I can put it back in the future should someone want to do another push like I did to drive down SafeHandle finalization debt.

Reply:
> This is irrespective of this change or this field, right? You're saying mono is expecting that handle is first and it may not be.

Yea, it's not required for this PR (all the other fields are non-reference so auto layout will get things right).  Just might as well do some future-proofing

Reply:
Overall I'm fine with the PR as is. It would be nice to get the layout attribute in here, but we could also do a separate PR for that.  If splitting the file feels more fragile to you @stephentoub , I'm ok with the PR as is.

Reply:
@lambdageek, mono actually already has StructLayout applied:
https://github.com/dotnet/runtime/blob/5e3dd8da34f1ffc3032197b3f9d52797431391c5/src/mono/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.Mono.cs#L6-L8